File: 0771-jewels-and-stones/0771-jewels-and-stones.cpp
class Solution {
public:
    int numJewelsInStones(string jewels, string stones) {
        int ans = 0;

        unordered_set<char> st;
        for (char c : jewels) st.insert(c);
        for (char stone : stones) 
            if (st.count(stone))
                ans++;
        return ans;
    }
};

File: 0771-jewels-and-stones/0771-jewels-and-stones.java
class Solution {
  public int numJewelsInStones(String jewels, String stones) {
    int ans = 0;
    char[] s = jewels.toCharArray();
    Set<Character> set = new HashSet<>();
    for (char c : s) set.add(c);
    for (char stone : stones.toCharArray())
      if (set.contains(stone))
        ++ans;

    return ans;
  }
}